enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,179 @@
+
+**Audit Log Rotation**
+
+The API server config provides 3 fields to dictate log rotation and retention policies which may be thinly exposed to the user via the MicroShift config API.  These fields are:
+
+* `audit-log-maxage` defines the maximum number of days to retain old audit log files.  MicroShift defaults to 10 days.
+* `audit-log-maxbackup` defines the maximum number of audit log files to retain. MicroShift defaults to 10 files.
+* `audit-log-maxsize` defines the maximum size in megabytes of the audit log file before it gets rotated
+
+**Non-standard Logging Path**
+
+The API server config provides a field to specify an alternative audit log file path.  This field may be exposed to the user via the MicroShift config API.  This field is:
+
+* `audit-log-path` defines the path to the audit log file.  MicroShift defaults to `/var/log/microshift/audit.log`
+
+**Policy Profiles**
+
+Policy profiles are defined as part of the `openshift-cluster-config-operator` API and are not currently enabled on MicroShift.  This means that duplication of logic between MicroShift and the operator are necessary to support the same functionality. This is not ideal, but is the best option available at this time.
+
+<-- TODO -->
+_Implementation Details Pending_
+
+## Design Details
+
+### Open Questions
+
+- Custom log paths: Can sos collect reports from dynamic paths? If not, can we symlink the log directory to as a means of faking a static path while writing logs to a custom location?  If so, can sos follow these symlinks?
+
+### Test Plan
+
+* Unit tests to validate new config APIs
+* Integration tests to verify configured audit logging policies are applied properly
+
+### Graduation Criteria
+
+* Upstream code, tests, docs merged
+* Downstream builds, docs updated
+* Automated CI coverage for new functionality
+* QE test plans defined
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation completed and published
+- Sufficient test coverage
+- Gather feedback from users
+- Available by default
+
+#### Tech Preview -> GA
+
+- Sufficient time for feedback
+- End-to-end tests
+
+## Implementation History
+
+n/a
+
+## Alternatives
+
+* Continue using hardcoded audit logging policy
```

Comment:
We also talked about using logrotate with host configuration, instead of relying on the built-in features. It would be good to discuss why that approach won't work, if that is the decision.

Reply:
+1 Will add